### ISSUE TITLE
Calculating progress when using spread

### ISSUE THREAD

**Issue by timpoultney** [kriskowal/q#704]
Hi, given an array of promises that all use notify(0-100), when using Q.spread the progress function calculation required to get a result in the range 0-100 is a bit tricky.

Is there a better answer than this:
`
progress = [[],[],[]];
Q.spread(promises, function( ...
  ...
}).progress(update) {
    progress[update.index].push(parseFloat(update.value));
    total = 0;
    progress.forEach(function(row) {
        for(i=row.length-1;i>-1;i--) {
            if(i>0) total += row[i]-row[i-1];
            else total += row[i];
        }
    })
    console.log("Progress: "+Math.round((total/(100*promises.length)*100))+"%";
})
`

And could the calculation become part of Q? Thanks.

**Comment by benjamingr** [comment]
Yes, check the 2.0 branch of Q for what's eventually replacing progress (estimation)
